### DISCUSSION TITLE
Would it be good also have freebsd based runner?

### DISCUSSION THREAD

**Discussion by rafael-santiago** [actions/virtual-environments#7914] [Installed software]
Why just only make available linux, windows and darwin as runner images? It sounds quite limited. I have one project that runs on Linux and FreeBSD, the annoy part is that github actions would attend me 100% easily if it would also offered me a FreeBSD runner, unfortunately I am thinking on moving to cirrus-ci because there I have linux and bsd. It is a pitty because by letting everything just in github it would make the things much more self-contained besides dropping out (external) dependencies (what for me is always awesome)… Maybe the github runner's pre-images scope should be more ambitious to attend more users that goes beyond only web stuff. Just saying…

**Comment by vpolikarpov-akvelon** [comment]
Hey @rafael-santiago. Thank you for your suggestion, but it's not easy though. Unfortunately, runner itself doesn't support FreeBSD and there are some reasons for it. Please read discussion here: https://github.com/actions/runner/issues/385. Also in that thread you may find suggestions and links to 3rd party solutions of how to workaround this limitation.

**Reply by rafael-santiago** [reply]
Thanks for replying @vpolikarpov-akvelon, I will take a look.

**Reply by emaste** [reply]
Note that the reasons enumerated in actions/runner#385 have been resolved, including dotnet support for FreeBSD.
